### ISSUE TITLE
Obtain function signatures from `@validate_call`

### ISSUE THREAD

**Issue by Dan-wanna-M** [pydantic/pydantic#9718]
### Initial Checks

- [X] I have searched Google & GitHub for similar requests and couldn't find anything
- [X] I have read and followed [the docs](https://docs.pydantic.dev) and still think this feature is missing

### Description

```python
@validate_call
def pos_or_kw(a: int, b: int = 2) -> str:
    return f'a={a} b={b}'

# print(pos_or_kw.arguments_info)
```
Is there a way to obtain function signature after using the decorator? For `BaseModel`, we can use `model_fields.` There seems not to exist equivalent properties for `@validate_call`. This will allow developers to use the pydantic-compatible version of arguments type extracting.

### Affected Components

- [ ] [Compatibility between releases](https://docs.pydantic.dev/changelog/)
- [ ] [Data validation/parsing](https://docs.pydantic.dev/concepts/models/#basic-model-usage)
- [ ] [Data serialization](https://docs.pydantic.dev/concepts/serialization/) - `.model_dump()` and `.model_dump_json()`
- [ ] [JSON Schema](https://docs.pydantic.dev/concepts/json_schema/)
- [ ] [Dataclasses](https://docs.pydantic.dev/concepts/dataclasses/)
- [ ] [Model Config](https://docs.pydantic.dev/concepts/config/)
- [ ] [Field Types](https://docs.pydantic.dev/api/types/) - adding or changing a particular data type
- [X] [Function validation decorator](https://docs.pydantic.dev/concepts/validation_decorator/)
- [ ] [Generic Models](https://docs.pydantic.dev/concepts/models/#generic-models)
- [ ] [Other Model behaviour](https://docs.pydantic.dev/concepts/models/) - `model_construct()`, pickling, private attributes, ORM mode
- [ ] [Plugins](https://docs.pydantic.dev/) and integration with other tools - mypy, FastAPI, python-devtools, Hypothesis, VS Code, PyCharm, etc.

**Comment by sydney-runkle** [comment]
@Dan-wanna-M,

Good point, there's not currently an easy way to do this. We attach minimal information to the wrapped function.

If we decide to change this, it'd be in `internal/_validate_call.py`.

**Comment by cisaacstern** [comment]
I have had a similar need and addressed it manually with, e.g.

```python
@validate_call
def pos_or_kw(a: int, b: int = 2) -> str:
    return f'a={a} b={b}'

annotations = {
  arg: annotation
  for arg, annotation in pos_or_kw.__annotations__.items()
}
print(annotations)  # --> {'a': int, 'b': int, 'return': str}
```

**Comment by Dan-wanna-M** [comment]
> I have had a similar need and addressed it manually with, e.g.
> 
> ```python
> @validate_call
> def pos_or_kw(a: int, b: int = 2) -> str:
>     return f'a={a} b={b}'
> 
> annotations = {
>   arg: annotation
>   for arg, annotation in pos_or_kw.__annotations__.items()
> }
> print(annotations)  # --> {'a': int, 'b': int, 'return': str}
> ```
Yes, this is probably the workaround I will use for now. The main problem is that this is not inherently pydantic-compatible since pydantic has its own `TypeInfo` and interoping between our `TypeInfo` and pydantic's could get quite nasty.

**Comment by nix010** [comment]
Will look into this

**Comment by VigneshVSV** [comment]
I would appreciate integrating this feature. 

I tried to subclass `ValidateCallWrapper` but still had a number of issues. 

I am working on a JSON representation for argument and return value signature of an RPC method call . I tried to add pydantic for the validation of argument and return values and retrieve the said JSON schema from pydantic. But it seems to be hard.
